service.go

```diff
@@ -294,17 +317,27 @@ func (svc *Service) hasPermission(app *App, event *nostr.Event, requestMethod st
 		// No permission for this request method
 		return false, NIP_47_ERROR_RESTRICTED, fmt.Sprintf("This app does not have permission to request %s", requestMethod)
 	}
+	ExpiresAt := appPermission.ExpiresAt
+	if !ExpiresAt.IsZero() && ExpiresAt.Before(time.Now()) {
+		svc.Logger.Info("This pubkey is expired")
+		return false, NIP_47_ERROR_EXPIRED, "This app has expired"
+	}
 	MaxAmountPerTransaction := appPermission.MaxAmountPerTransaction
 	if MaxAmountPerTransaction != 0 {
 		if paymentRequest.MSatoshi/1000 > int64(MaxAmountPerTransaction) {
 			svc.Logger.Info("trying to send more than max amount per transaction in budget")
 			return false, NIP_47_ERROR_INSUFFICIENT_BALANCE, "Payment amount is greater than budget allows"
 		}
 	}
+	budgetType := appPermission.BudgetType
 	maxAmount := appPermission.MaxAmount
 	if maxAmount != 0 {
 		var result SumResult
-		svc.db.Table("payments").Select("SUM(amount) as sum").Where("app_id = ? AND preimage IS NOT NULL AND created_at > ?", app.ID, time.Now().AddDate(0, -1, 0)).Scan(&result)
+		if budgetType == "never" {
+			svc.db.Table("payments").Select("SUM(amount) as sum").Where("app_id = ? AND preimage IS NOT NULL", app.ID).Scan(&result)
```

Comment:
Maybe this should still use GetStartOfBudget, but just return the creation date of the app connection if the type is "never". Then we only have one query